icu4c/source/test/cintltst/ccaltst.c

```diff
@@ -1631,8 +1632,12 @@ static void TestGetKeywordValuesForLocale() {
             { "zh_TW",       "gregorian", "roc", "chinese", NULL, NULL },
             { "ar_IR",       "persian", "gregorian", "islamic", "islamic-civil", "islamic-tbla" },
             { "th@rg=SAZZZZ", "islamic-umalqura", "gregorian", "islamic", "islamic-rgsa", NULL },
+        
+            // test for ICU-22364
+            { "zh_CN@rg=TWzzzz", "gregorian", "roc", "chinese", NULL, NULL },
+            { "zh_CN@rg=TWxxxx", "gregorian", "roc", "chinese", NULL, NULL },
```

Comment:
TWxxxx is not listed in https://github.com/unicode-org/cldr/blob/main/common/supplemental/subdivisions.xml